tests/Unit/VehicleTest.php

```diff
@@ -34,4 +33,17 @@ public function testScopeEstado()
     $this->assertEquals($vehiculoModel->first()->estado, $vehiculo->estado);
     $this->assertEquals($vehiculoModel->count(),1);
   }
+  public function testVehiculos()
+  {
+    $service = factory(Servicio::class)->create();
+    $vehiculo = factory(Vehiculo::class)->create();
+    $vehiculoServicio = factory(VehiculoServicio::class)->create(['vehiculo_id'=>$service->id]);
+    $this->assertEquals($vehiculoServicio->first()->vehiculo_id,$service->id);
```

Comment:
Esto y el resto estan mal. El test tiene que probar el scope que esta en Vehiculos.
![image](https://user-images.githubusercontent.com/874682/33495895-f0f27586-d6a6-11e7-829a-d7ecb5736c3e.png)
Vos aca y en el resto estas probando vehiculoServicio, que es como instanciar una variable y ver si se instancio bien. El assert tiene que ser:
```
$this->assertEquals($vehiculo->servicios()->fist->service_id,$service->id);
```

Reply:
com el codigo asi 

public function testServicios()
  {
    $service = factory(Servicio::class)->create();
    $vehiculo = factory(Vehiculo::class)->create();
    $this->assertEquals($vehiculo->servicios()->first->service_id,$service->id);
  }

me tira este error 

Tests\Unit\VehicleTest::testServicios
ErrorException: Undefined property: Illuminate\Database\Eloquent\Relations\HasMany::$first

Y lo mismo con el test de vehiculo

Reply:
Y si le pueden faltar algunos parentesis a lo que escribi, no lo probe, querias que entiendas que estabas probando algo que no tiene sentido :).
Me parece que el problema es que el first lleva parentesis
```
$this->assertEquals($vehiculo->servicios()->first()->service_id,$service->id);
```

Reply:
Claro, me imagine que era algo de eso jajaja, el de vehiculo-material pude hacerlo andar... pero el de vehiculo-servicio no... este es el codigo que use:

public function testServicios()
  {
    $vehiculo = factory(Vehiculo::class)->create();
    $service = factory(Servicio::class)->create();
    $this->assertEquals($vehiculo->servicios()->first()->servicio_id,$service->id);
  }

este es el error que me tira:

1) Tests\Unit\VehicleTest::testServicios
ErrorException: Trying to get property of non-object

Reply:
Te falta crear la relacion entre vehuiculo y servicio.  Algo asi:
```
$vehiculoServicio = factory(VehiculoServicio::class)->create(['vehiculo_id'=>$service->id]);
```

Reply:
Estaba probando justo eso si, pero me tira el mismo error, creo que no lo estoy metiendo en el assertEquals, porque asi  $this->assertEquals($vehiculoServicio->servicios()->first()->servicio_id,$service->id); 
no funciona porque esta buscandolo dentro de vehiculoServicio, entonces no me ubico

Reply:
Donde esta el metodo que queres testear?

Reply:
en vehiculo, pero asi
 $this->assertEquals($vehiculo->servicios()->first()->servicio_id,$service->id);
me tira  
ErrorException: Trying to get property of non-object
por mas que ya cree vehiculoServicio... 
Por eso pregunto si hay que asignarlo tambien en algun lugar del aassertEquals

Reply:
Hola @SSAAMMAA me baje el codigo e hice los test de Vehiculo. a mi me funcionan perfecto con los cambios que te dije:
```
public function testVehiculos()
  {
    $service = factory(Servicio::class)->create();
    $vehiculo = factory(Vehiculo::class)->create();
    $vehiculoServicio = factory(VehiculoServicio::class)->create(['vehiculo_id'=>$service->id]);
    $this->assertEquals($vehiculo->servicios()->first()->vehiculo_id, $service->id);
    $this->assertEquals($vehiculo->servicios()->count(), 1);
  }
  public function testMateriales()
  {
    $vehiculo = factory(Vehiculo::class)->create();
    $material = factory(Material::class)->create(['vehiculo_id'=>$vehiculo->id]);
    $this->assertEquals($vehiculo->materiales()->first()->material_id, $material->material_id);
    $this->assertEquals($vehiculo->materiales()->count(), 1);
  }
```

Reply:
@SSAAMMAA hace lo mismo para los otros tests y estariamos listos para mergear.
Ademas tenes que sincronizarte el branch.